src/main/java/com/gachigang/ontherun/service/AuditorAwareImpl.java

```diff
@@ -0,0 +1,34 @@
+package com.gachigang.ontherun.service;
+
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+/**
+ * Provides the current auditor for auditing in a Spring Security-enabled application.
+ *
+ * <p>
+ * This service retrieves the current authentication information from the Spring Security context
+ * to determine the current user.
+ *
+ * @see AuditorAware
+ */
+@Service
+public class AuditorAwareImpl implements AuditorAware<String>  {
+
+
+    // TODO: Review Spring Security configuration to resolve "anonymousUser" issue.
+    @Override
+    public Optional<String> getCurrentAuditor() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication != null && authentication.isAuthenticated()) {
```

Comment:
Need refactoring, it is a bad approach to have a return statement in if and else. You can extract return statement from if-else, and I believe that terminal operator will look better here